java-frontend/src/main/java/org/sonar/java/UCFGJavaVisitor.java

```diff
@@ -303,6 +281,30 @@ private void buildMethodInvocation(UCFGBuilder.BlockBuilder blockBuilder, Identi
     }
   }
 
+  private void buildPlusOrAssignmentInvocation(BlockBuilder blockBuilder, IdentifierGenerator idGenerator, Tree element) {
+    boolean elementIsString = isString(((ExpressionTree) element).symbolType());
+    if (element.is(PLUS) && elementIsString) {
+      BinaryExpressionTree binaryExpressionTree = (BinaryExpressionTree) element;
+      Expression lhs = idGenerator.lookupExpressionFor(binaryExpressionTree.leftOperand());
+      Expression rhs = idGenerator.lookupExpressionFor(binaryExpressionTree.rightOperand());
+      Expression.Variable var = variableWithId(idGenerator.newIdFor(binaryExpressionTree));
+      blockBuilder.assignTo(var, call("__concat").withArgs(lhs, rhs), location(element));
+    } else if (element.is(PLUS_ASSIGNMENT) && elementIsString) {
+      Expression var = idGenerator.lookupExpressionFor(((AssignmentExpressionTree) element).variable());
+      Expression expr = idGenerator.lookupExpressionFor(((AssignmentExpressionTree) element).expression());
+      if (!var.isConstant()) {
```

Comment:
this condition was used to safely cast to Variable, but with the upgrade to latest version of protobuf this is not enough to guarantee that var will be an Expression.Variable.

Reply:
I don't know how to cover that with tests, though